apps/examples/testcase/le_tc/network/tc_net_mac.c

```diff
@@ -29,269 +29,153 @@
 * @brief					:
-* @scenario					:   
+* @scenario					:
 * @apicovered				: net_clone
 * @precondition				:
 * @postcondition			:
 */
-static void tc_net_clone_p(void)
+static void tc_net_clone_p(int sockfd1, int sockfd2)
 {
-	int sockfd1;
-	int sockfd2;
 	int result;
 	struct socket *sock1 = NULL;
 	struct socket *sock2 = NULL;
-	sockfd1 = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
-	if (sockfd1 == -1) {
-		return ;
-	}
-	sockfd2 = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
-	if (sockfd2 == -1) {
-		return ;
-	}
+
 	sock1 = get_socket(sockfd1);
-	if (!sock1) {
-		return ;
-	}
+	TC_ASSERT_NEQ("get_socket", sock1, NULL);
 	sock2 = get_socket(sockfd2);
-	if (!sock2) {
-		return ;
-	}
-	
-	result = net_clone(sock1, sock2);
-	TC_ASSERT_EQ("tc_net_clone_p", result, 0);
-	TC_SUCCESS_RESULT();
-}
+	TC_ASSERT_NEQ("get_socket", sock2, NULL);
 
-/**
-* @statitcase				: tc_net_clone_n
-* @brief					:
-* @scenario					:   
-* @apicovered				: net_clone
-* @precondition				:
-* @postcondition			:
-*/
-static void tc_net_clone_n(void)
-{
-	struct socket *sock1 = NULL;
-	struct socket *sock2 = NULL;
-	int result;
-	
 	result = net_clone(sock1, sock2);
-	TC_ASSERT_NEQ("tc_net_clone_n", result, 0);
+	TC_ASSERT_EQ("tc_net_clone_p", result, ZERO);
 	TC_SUCCESS_RESULT();
 }
 
 /**
 * @statitcase				: tc_net_dupsd2_p
 * @brief					:
-* @scenario					:   
+* @scenario					:
 * @apicovered				: net_dupsd2
 * @precondition				:
 * @postcondition			:
 */
-static void tc_net_dupsd2_p(void)
+static void tc_net_dupsd2_p(int old_fd, int new_fd)
 {
-	int old_sockfd;
-	int new_sockfd;
 	int result;
-	old_sockfd = 3;
-	new_sockfd = 5;
 
-	result = net_dupsd2(old_sockfd, new_sockfd);
-	TC_ASSERT_EQ("tc_net_dupsd2_p", result, 0);
+	result = net_dupsd2(old_fd, new_fd);
+	TC_ASSERT_EQ("tc_net_dupsd2_p", result, ZERO);
 	TC_SUCCESS_RESULT();
 }
 
 /**
 * @statitcase				: tc_net_dupsd2_n
 * @brief					:
-* @scenario					:   
+* @scenario					:
 * @apicovered				: net_dupsd2
 * @precondition				:
 * @postcondition			:
 */
-static void tc_net_dupsd2_n(void)
+static void tc_net_dupsd2_n(int old_fd, int new_fd)
 {
-	int old_sockfd;
-	int new_sockfd;
 	int result;
-	old_sockfd = -1;
-	new_sockfd = 5;
 
-	result = net_dupsd2(old_sockfd, new_sockfd);
-	TC_ASSERT_NEQ("tc_net_dupsd2_n", result, 0);
+	result = net_dupsd2(old_fd, new_fd);
+	TC_ASSERT_EQ("tc_net_dupsd2_n", result, NEG_VAL);
 	TC_SUCCESS_RESULT();
 }
 
 /**
 * @statitcase				: tc_net_ethernetif_status_callback_p
 * @brief					:
-* @scenario					:   
+* @scenario					:
 * @apicovered				: ethernetif_status_callback
 * @precondition				:
 * @postcondition			:
 */
 static void tc_net_ethernetif_status_callback_p(void)
 {
 	struct netif *netif = NULL;
-	if (!(netif = (struct netif *)malloc(sizeof(struct netif))))
-		return;
+	netif = (struct netif *)malloc(sizeof(struct netif));
+	TC_ASSERT_NEQ("malloc", netif, NULL);
 
 	ethernetif_status_callback(netif);
+	TC_FREE_MEMORY(netif);
 	TC_SUCCESS_RESULT();
```

Comment:
This tc is ethernetif_status_callback. But you didn't check.

Reply:
It is only printing IP address.